packages/flutter_tools/lib/src/desktop_device.dart

```diff
@@ -202,6 +204,92 @@ abstract class DesktopDevice extends Device {
   /// Called after a process is attached, allowing any device-specific extra
   /// steps to be run.
   void onAttached(ApplicationPackage package, BuildMode buildMode, Process process) {}
+
+  /// Computes a set of environment variables used to pass debugging information
+  /// to the engine without interfering with application level command line
+  /// arguments.
+  ///
+  /// The format of the environment variables is:
+  ///   * FLUTTER_ENGINE_SWITCHES to the number of switches.
+  ///   * FLUTTER_ENGINE_SWITCH_<N> (indexing from 1) to the individual switches.
+  Map<String, String> _computeEnvironment(DebuggingOptions debuggingOptions, bool traceStartup, String route) {
+    int flags = 1;
```

Comment:
Wouldn't it be clearer to start this as zero, increment flags at the start of `addFlag` instead of the end, and not have to subtract in `finish`?

Reply:
done